app.py

```diff
@@ -1,24 +1,159 @@
-from flask import url_for,render_template,redirect,Flask,flash
-from forms import LoginForm
+from flask import Flask, render_template, flash, redirect, url_for, session, request, logging, jsonify
+from passlib.hash import sha256_crypt
+from functools import wraps
+from forms import RegisterForm, LoginForm
 from Make_Tables.mysqlconnect import mydb, mycursor, create_insert_statement   #Imported the mysqlconnect.py file from Make_tables folder
+import gc
+
+
 
 app=Flask(__name__,static_url_path='/public')
 app.config['SECRET_KEY']='c828b6ff21f45063fd7860e5c1b1d233'
+app.config['MYSQL_CURSORCLASS'] = 'DictCursor'
+
 
+# Home page before logging in
 @app.route('/')
 def home():
-    return render_template('index.html')
-
-@app.route("/login",methods=['GET','POST'])
-def Login():
-    form=LoginForm()
-    if form.validate_on_submit():
-        if form.email.data[1:6].isdecimal() and form.email.data[6:]=='@students.iitmandi.ac.in' :
-            flash('You have logged-in successfully!')
-            return redirect(url_for('home'))
+    return 'Home page of SAC-Portal.'
+
+# Route guard to prevent logged in user access to login and register page
+def not_logged_in(f):
```

Comment:
- [ ] move not_logged in and logged in helper to another file
- [ ] squash all commits into one
- [ ] add a .gitignore